### PR TITLE
Init README Deploy Updates

### PR DESCRIPTION
**Pull request by awood45**
*Issue #, if available:*

*Description of changes:*

*Checklist:*

- [ ] Write Design Document ([Do I need to write a design document?](https://github.com/awslabs/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.rst#design-document))
- [ ] Write unit tests
- [ ] Write/update functional tests
- [ ] Write/update integration tests
- [ ] `make pr` passes
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samcli/local/init/templates/cookiecutter-aws-sam-hello-golang/{{cookiecutter.project_name}}/README.md [hunk 1]
```diff
@@ -77,38 +77,21 @@ AWS Lambda Python runtime requires a flat folder with all dependencies including
             ...
 ```
 
-First and foremost, we need a `S3 bucket` where we can upload our Lambda functions packaged as ZIP before we deploy anything - If you don't have a S3 bucket to store code artifacts then this is a good time to create one:
+To deploy your application for the first time, run the following in your shell:
 
 ```bash
-aws s3 mb s3://BUCKET_NAME
+sam deploy --guided
```
Comment (c2tarun) [inline samcli/local/init/templates/cookiecutter-aws-sam-hello-golang/{{cookiecutter.project_name}}/README.md]: Do we need `sam build` here?

File: samcli/local/init/templates/cookiecutter-aws-sam-hello-golang/{{cookiecutter.project_name}}/README.md [hunk 2]
```diff
@@ -77,38 +77,21 @@ AWS Lambda Python runtime requires a flat folder with all dependencies including
...
 ```
 
-Next, run the following command to package our Lambda function to S3:
+The first command will build the source of your application. The second command will package and deploy your application to AWS, with a series of prompts:
```
Comment (c2tarun) [inline samcli/local/init/templates/cookiecutter-aws-sam-hello-golang/{{cookiecutter.project_name}}/README.md]: Only one command mentioned above.

Comment (awood45) [reply samcli/local/init/templates/cookiecutter-aws-sam-hello-golang/{{cookiecutter.project_name}}/README.md]: Addressed in change

File: samcli/local/init/templates/cookiecutter-aws-sam-hello-golang/{{cookiecutter.project_name}}/README.md (hunk 1 above)
Comment (awood45) [reply samcli/local/init/templates/cookiecutter-aws-sam-hello-golang/{{cookiecutter.project_name}}/README.md]: No, Go build is not yet supported. Once it is, we would add it there.

File: samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs/{{cookiecutter.project_name}}/README.md [hunk 3]
```diff
@@ -23,44 +23,26 @@ The Serverless Application Model Command Line Interface (SAM CLI) is an extensio
...
-    --s3-bucket BUCKET_NAME
-```
-
-The SAM CLI creates deployment packages, uploads them to the S3 bucket, and creates a new version of the template that refers to the artifacts in the bucket. 
+* **Stack Name**: The name of the stack to deploy to CloudFormation. This should be unique to your account and region, and a good starting point would be something matching your project name.
+* **AWS Region**: The AWS region you want to deploy your app to.
+* **Confirm changes before deploy**: If set to yes, any change sets will be shown to you before execution for manual review. If set to no, the AWS SAM CLI will automatically deploy application changes.
+* **Allow SAM CLI IAM role creation**: Many AWS SAM templates, including this example, create AWS IAM roles required for the AWS Lambda function(s) included to access AWS services. By default, these are scoped down to minimum required permissions. To deploy an AWS CloudFormation stack which creates or modified IAM roles, the `CAPABILITY_IAM` parameter must be provided. If permission isn't provided through this prompt, to deploy this example you must explicitly pass `--capabilities CAPABILITY_IAM` to the `sam deploy` command.
```
Comment (TheSriram) [inline samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs/{{cookiecutter.project_name}}/README.md]: > the `CAPABILITY_IAM` parameter must be provided

 `--capabilities` parameter must be provided with a value of `CAPABILITY_IAM`

Comment (awood45) [reply samcli/local/init/templates/cookiecutter-aws-sam-hello-nodejs/{{cookiecutter.project_name}}/README.md]: Changed wording

**Pull request merged**